runner/src/indexer/indexer.ts

```diff
@@ -13,14 +13,22 @@ import type IndexerConfig from '../indexer-config';
 import { type PostgresConnectionParams } from '../pg-client';
 import IndexerMeta, { IndexerStatus } from '../indexer-meta';
 import { wrapSpan } from '../utility';
+import { type IRpcClient } from '../rpc-client/rpc-client';
+import { type CodeResult } from '@near-js/types/lib/provider/response';
 
 interface Dependencies {
   fetch: typeof fetch
   provisioner: Provisioner
   dmlHandler?: DmlHandler
   indexerMeta?: IndexerMeta
   parser: Parser
-};
+  rpcClient?: IRpcClient
```

Comment:
```suggestion
  rpcClient: IRpcClient
```
I'm not sure there is any reason not to include this? `dmlHandler` etc is only optional because we need to get DB credentials. 

This would also tidy up the `context` generation since it would always be defined.